sopel/plugin.py

```diff
@@ -2029,6 +2030,6 @@ def add_attribute(function):
     return add_attribute
 
 
-def abort(*args, **kwargs) -> None:
+def abort(*args: Any, **kwargs: Any) -> None:
```

Comment:
I can see room for an argument that the API of `PluginAbort` should be restricted to exactly two valid invocations: `PluginAbort()` and `PluginAbort(msg: str)` but I wired this up as truly arbitrary and I hate that I had to annotate `Any` here.

Reply:
IMO it's entirely reasonable to insist on providing One Common Pattern (`abort(msg: str | None)`) through this helper, and make plugin devs raise the exception themselves if they want something non-standard.

Especially because the intent of this feature, as I understand it, is as a substitute for `bot.reply("Whoops!"); return`. Supporting more than an optional abort message seems… "extra" of you 🤔